### ISSUE TITLE
`no-new-object` to allow argument

### ISSUE THREAD

**Issue by brettz9** [eslint/eslint#11810]
**What rule do you want to change?**

`no-new-object`

**Does this change cause the rule to produce more or fewer warnings?**

Fewer warnings.

**How will the change be implemented? (New option, new default behavior, etc.)?**

New default behavior.

**Please provide some example code that this change will affect:**

```js
new Object(BigInt('1n'))
```

**What does the rule currently do for this code?**

Reports an error.

**What will the rule do after it's changed?**

Not report anything.

**Are you willing to submit a pull request to implement this change?**

Yes.

As per the above example, there are [uses](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/BigInt#Type_information) for wrapping some items within a `new Object()` (or within the equivalent, `Object()`, but some prefer the style of requiring `new` in such cases, as per [`unicorn/new-for-builtins`](https://github.com/sindresorhus/eslint-plugin-unicorn/blob/master/docs/rules/new-for-builtins.md)).

If it is thought that people will like to make this optional, I can modify to something like `allowWithArguments`.

**Comment by platinumazure** [comment]
This should definitely be optional. :smile:

**Comment by mysticatea** [comment]
Out of curiosity, why do you want to use the wrapper objects?

**Comment by brettz9** [comment]
I've changed it to be optional and off-by-default (and switching the commit message from "Breaking" to "New").

My own need is merely for [typeson-registry](https://github.com/dfahlander/typeson-registry/)'s ability to serialize/deserialize `BigInt` objects in case anyone uses them within the structured cloning algorithm. And we're using `typeson-registry` with [IndexedDBShim](https://github.com/axemclion/IndexedDBShim/) to allow it properly store such objects. And I was motivated to do this primarily because the W3C tests include this use case. :)

But I think some may wish to use this for polyfilling or extending the prototype. Interestingly, `Object()` converts to a corresponding object, e.g., BigInt, Number, Boolean, String, so it is not just for polyfilling/enhancing `Object`. In the case of `BigInt`, therefore, one could add special arithmetic methods on `BigInt.prototype`. For a dumb example:

```js
BigInt.prototype.plusOne = function () {
    return this + 1n;
};

Object(1n).plusOne();
```

**Comment by mysticatea** [comment]
Because JS runtime does boxing automatically, so you don't need to wrap it by `Object()`. I.e., `1n.plusOne()` works as expected.

**Comment by brettz9** [comment]
Btw, the reason I originally suggested the behavior as the default is because the only rationale given in the docs for avoiding `new Object` is that it is equivalent to the object literal, but in this case it isn't.

However, I understand people don't always want to allow such more obscure uses.

**Comment by brettz9** [comment]
Ah, ok, I see, thanks, @mysticatea . I've amended the docs to avoid mentioning that. But it is still possible to use it. I've also seen `Object()` used to polyfill certain built-in methods, e.g., https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/find#Polyfill (as part of `ToObject`), though this seems pretty obscure here too--I am not sure what case this covers, however.
